### PR TITLE
Move device code in get_json_object to cu or cuh

### PR DESCRIPTION
**Pull request by thirtiseven**
Contributes to https://github.com/NVIDIA/spark-rapids-jni/issues/1906

This PR:
- Move Kernel code to cuh file.
- Expose MAX_PATH_DEPTH in JSONUtils.java (will call it from plugin in a plugin PR).

I think we can make this moving code change first so it will be easy to review when addressing other comments.

### CODE REVIEW

File: src/main/java/com/nvidia/spark/rapids/jni/JSONUtils.java [hunk 1]
```diff
@@ -23,6 +23,8 @@ public class JSONUtils {
     NativeDepsLoader.loadNativeDeps();
   }
 
+  public static final int MAX_PATH_DEPTH = 32;
```
Comment (res-life) [inline src/main/java/com/nvidia/spark/rapids/jni/JSONUtils.java:26]: I did not find any usage for this new variable.

Comment (thirtiseven) [reply src/main/java/com/nvidia/spark/rapids/jni/JSONUtils.java:26]: It will be used in plugin, for max path instructions fallback.

**Review by res-life — APPROVED** [review]

**Comment by thirtiseven** [comment]
build

**Comment by thirtiseven** [comment]
build

**Review by res-life — APPROVED** [review]

**Pull request merged**